deta/base.py

```diff
@@ -148,23 +162,20 @@ def fetch(
         self,
         query: typing.Union[dict, list] = None,
         *,
-        buffer: int = None,
-        pages: int = 10,
-    ) -> typing.Generator:
+        limit: int = 1000,
+        last: str = None,
+    ):
         """
         fetch items from the database.
             `query` is an optional filter or list of filters. Without filter, it will return the whole db.
-        Returns a generator with all the result, We will paginate the request based on `buffer`.
         """
-        last = True
-        code = 200
-        counter = 0
-        while code == 200 and last and pages > counter:
-            code, res = self._fetch(query, buffer, last)
-            items = res["items"]
-            last = res["paging"].get("last")
-            counter += 1
-            yield items
+        code, res = self._fetch(query, limit, last)
+        if code != 200:
+            return None
```

Comment:
not sure if we should return none or a `FetchResponse` with `count = 0`, `last = None` and ` items = []`